tests/baselines/reference/unknownControlFlow.types

```diff
@@ -819,55 +819,55 @@ function fx1<T extends unknown>(value: T & ({} | null)) {
 
 function fx2<T extends {}>(value: T & ({} | null)) {
 >fx2 : <T extends {}>(value: T & ({} | null)) => void
->value : T & ({} | null)
+>value : T
 
     if (value === 42) {
 >value === 42 : boolean
->value : T & ({} | null)
+>value : T
 >42 : 42
 
         value;  // T & {}
->value : T & ({} | null)
+>value : never

```

Comment:
This seems funky; `T` could be `42` here but it's thinking that comparison is impossible and made `never`.

Reply:
It's actually existing behavior that is being revealed by the reduction of `T & ({} | null)` to just `T`. We have logic somewhere in the comparable relation that considers a naked type parameter constrained to `{}` to be something that always represents an object.